File: ComputationalGeometry/pagina02/1549/1549.php
<?php

$n = (int) readline();
for($i = 0; $i < $n; $i++){

    list($pessoas, $mililitros) = explode(" ", readline());
    list($b, $B, $H) = explode(" ", readline());

    $b = (int) $b; $B = (int) $B; $H = (int) $H;

    $media = (int) $mililitros / (int) $pessoas;
    $temp = pow(($media * 3 * ($B - $b) / (pi() * $H)) + pow($b, 3), (1/3));
    $h = $media * 3 / (pi() * (pow($temp, 2) + $temp * $b + pow($b, 2)));

    echo number_format($h, 2, ".", "") . "\n";
}

File: ComputationalGeometry/__1000__/1549/1549.cpp
#include <iostream>
#include <numbers>   // std::numbers::pi (C++20)
#include <cmath>     // pow, cbrt
#include <iomanip>   // setprecision, fixed

using namespace std;

int main() {
    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {
        int pessoas, mililitros, b, B, H;
        cin >> pessoas >> mililitros >> b >> B >> H;

        float media = static_cast<float>(mililitros) / pessoas;
        float temp = cbrt((media * 3 * (B - b) / (numbers::pi * H)) + pow(b, 3));
        float h = media * 3 / (numbers::pi * (pow(temp, 2) + temp * b + pow(b, 2)));

        cout << fixed << setprecision(2) << h << endl;
    }

    return 0;
}